democracy/factories/comment.py

```diff
@@ -13,6 +13,8 @@ class BaseCommentFactory(factory.django.DjangoModelFactory):
 
     @factory.post_generation
     def post(self, create, extracted, **kwargs):
+        if extracted:
+            return
```

Comment:
I think I get the idea here (you want to disable the post-hook from running if `post` has any value set), but it's very confusing... and so is the original implementation. How this should look like is something like this:

```py
# First of all, just remove the BaseCommentFactory and fully implement SectionCommentFactory instead
class SectionCommentFactory(factory.django.DjangoModelFactory):
    created_by = factory.LazyAttribute(lambda obj: get_random_user())
    content = factory.Faker("text")

    class Meta:
        model = SectionComment

    @factory.post_generation
    def set_created_at_in_future(self, create, extracted, **kwargs):
        # NOTE: I really hate keeping this part of the code, but I've no clue if removing this would break any tests or not. Remove if not.
        # Can't be done in a lazy attribute because they have no access to the concrete factory's model class,  # noqa: E501
        # for `parent_field`...
        self.created_at = getattr(self, self.parent_field).created_at + timedelta(
            seconds=random.randint(120, 600)
        )
        self.save(update_fields=("created_at",))


class SectionCommentWithVotes(SectionCommentFactory):
    @factory.post_generation
    def add_random_voters(self, create, extracted, **kwargs):
        # NOTE: This is terrible code and makes the tests *not* deterministic, but I'm 99% sure removing this would break a lot of tests.
        for _x in range(int(random.random() * random.random() * 5)):
            self.voters.add(get_random_user())
        self.recache_n_votes()

# In a normal, sane test...
def test_foo():
    my_comment = SectionCommentFactory(...)
    ...

# In every existing test in this project:
def test_bar():
    my_comment_with_votes = SectionCommentWithVotesFactory(...)
    ...
```

Of course, this solution requires you to rename a lot of things (namely, `SectionCommentFactory` -> `SectionCommentWithVotesFactory`, don't need to deal with variable names etc.). However, it also makes it more clear what's going on in the future. I think it's worth doing.

Reply:
Yes, the refactoring would pay itself in the future. In the process of fighting with the tests the idea of refactoring bunch of unrelated tests seemed a deeper pithole.

Reply:
Will make a ticket out of this.